salt/states/pkgrepo.py

```diff
@@ -390,6 +398,21 @@ def managed(name, ppa=None, copr=None, aptkey=True, **kwargs):
         # If neither argument was passed we assume the repo will be enabled
         enabled = True
 
+    # To be changed in version 3008: default to False and still log a warning
+    allow_insecure_key = kwargs.pop("allow_insecure_key", True)
+    if allow_insecure_key:
+        salt.utils.versions.warn_until(
```

Comment:
I think we should only warn if someone is using an insecure key url. This will get pretty noisy for users that don't need to update their states when 3008 is released because they are using secure url's.